btree/btree.c

```diff
@@ -1962,9 +1955,16 @@ static int64_t node_alloc(struct node_op *op, struct td *tree, int size,
 			  const struct node_type *nt, int ksize, int vsize,
 			  struct m0_be_tx *tx, int nxt)
 {
-	int        nxt_state;
+	int        nxt_state = nxt;
+	void      *area;
+	int        actual_size = 1ULL << size;
 
-	nxt_state = segops->so_node_alloc(op, tree, size, nt, tx, nxt);
+	M0_PRE(op->no_opc == NOP_ALLOC);
+	M0_PRE(node_shift_is_valid(size));
+	area = m0_alloc_aligned(actual_size, size);
+	M0_ASSERT(area != NULL);
+	op->no_addr = segaddr_build(area, size);
+	op->no_tree = tree;
```

Comment:
put it in a function say 'seg_node_alloc`

Reply:
I bought these changes in node_alloc as we wanted to remove the dependency of a mem_node_*() function.
I will make a new function to call this block of code.